src/client.rs

```diff
@@ -131,34 +95,24 @@ impl Client {
     /// ```
 
     pub fn with_http_client(http_client: Arc<dyn HttpClient>) -> Self {
-        Self {
+        let client = Self {
             base_url: None,
             http_client,
-            middleware: Arc::new(vec![
-                #[cfg(feature = "middleware-logger")]
-                Arc::new(crate::middleware::Logger::new()),
-            ]),
-        }
+            middleware: Arc::new(vec![]),
+        };
+
+        #[cfg(feature = "middleware-logger")]
+        let client = client.with(crate::middleware::Logger::new());
```

Comment:
Why was this moved out to here rather than just being in the vec?

Reply:
I found it confusing to read a feature flag in that position, and thought other people might also. I think rust-analyzer also complained about it in that position

Reply:
what's the argument against using `with`? another argument for it is that if the logic for what needed to be done in order to insert a middleware changed, there would only be one place that had to change

Reply:
This is also how tide does default middleware: https://github.com/http-rs/tide/blob/main/src/server.rs#L100-L102